qiskit_braket_provider/providers/adapter.py

```diff
@@ -376,24 +427,42 @@ def to_braket(circuit: QuantumCircuit) -> Circuit:
             raise NotImplementedError(
                 "reset operation not supported by qiskit to braket adapter"
             )
-        else:
-            params = operation.params if hasattr(operation, "params") else []
-
-            for i, param in enumerate(params):
-                if isinstance(param, Parameter):
-                    params[i] = FreeParameter(param.name)
+        elif gate_name in _qiskit_controlled_gate_names_to_braket_gates:
+            params = _create_free_parameters(operation)
+
+            gate = _qiskit_controlled_gate_names_to_braket_gates[gate_name](*params)
+            qubit_indices = [circuit.find_bit(qubit).index for qubit in qubits]
+            gate_qubit_count = gate.qubit_count
+            target_indices = qubit_indices[-gate_qubit_count:]
+            instruction = Instruction(
+                # Getting the index from the bit mapping
+                operator=gate,
+                target=target_indices,
+                control=qubit_indices[:-gate_qubit_count],
```

Comment:
I'm going to hold off on `ctrl_state` for the time being, because that and the handling of negctrl deserves its own PR. Technically, we're _already_ handling controlled gates incorrectly in the provider; in Qiskit, the control state can only come from control qubits that are part of the gate, while Braket only allows the control state for _additional_ controlled qubits. For example, the `CCX` gate can have a control state from any of {0, 1, 2, 3} in Qiskit, but can't have _any_ control state in Braket unless it's modified with additional control qubits.

Reply:
I tried to do that, but the behavior can't be forced without mocking, since the transpiled circuit will never have any negative control. I've added it back, this time mocking out `transpile` in the test.